java/ql/lib/semmle/code/java/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -234,6 +234,17 @@ predicate storeStep(Node node1, ContentSet f, Node node2) {
     pragma[only_bind_out](node2.getEnclosingCallable())
 }
 
+private Field getLexicallyOrderedRecordField(Record r, int idx) {
+  result =
+    rank[idx + 1](Field f, int i, Parameter p |
+      f = r.getAField() and
+      p = r.getCanonicalConstructor().getParameter(i) and
+      f.getName() = p.getName()
```

Comment:
I guess this is unlikely to be a performance problem any time soon, but as-is it's guaranteed to be suboptimal, since there's no efficient join-ordering as written.